src/Network.cpp

```diff
@@ -930,45 +930,33 @@ Network::Netresult Network::get_scored_moves_internal(
     // Sigmoid
     const auto winrate_sig = (1.0f + std::tanh(winrate_out[0])) / 2.0f;
 
-    std::vector<scored_node> result;
+    Netresult result;
+
     for (auto idx = size_t{0}; idx < outputs.size(); idx++) {
         if (idx < BOARD_SQUARES) {
             const auto rot_idx = rotate_nn_idx_table[rotation][idx];
-            const auto x = rot_idx % BOARD_SIZE;
-            const auto y = rot_idx / BOARD_SIZE;
-            const auto rot_vtx = state->board.get_vertex(x, y);
-            if (state->board.get_square(rot_vtx) == FastBoard::EMPTY) {
-                result.emplace_back(outputs[idx], rot_vtx);
-            }
+            result[rot_idx] = outputs[idx];
         } else {
-            result.emplace_back(outputs[idx], FastBoard::PASS);
+            result[idx] = outputs[idx];
         }
     }
+    result[BOARD_SQUARES+1] = winrate_sig;
```

Comment:
This is very ugly. At least make it a pair of an array and a float, or define a type/class/struct for the result.